### ISSUE TITLE
unified-signatures: rest parameter overloading

### ISSUE THREAD

**Issue by WilcoBakker** [palantir/tslint#2845]
### Bug Report

- __TSLint version__: 5.3.2
- __TypeScript version__: 2.3.3
- __Running TSLint via__: Atom

#### TypeScript code being linted

```ts
protected oneOf(...accepted: string[]): (arg1: string, arg2: any) => void;
protected oneOf(...accepted: number[]): (arg1: string, arg2: any) => void;
```

with `tslint.json` configuration:

```json
{
  "extends": "tslint:recommended",
  "rules": {
    "quotemark": "double",
    "no-console": false,
    "jsdoc-format": false,
    "interface-name": false,
    "object-literal-sort-keys": false,
    "no-bitwise": false
  }
}
```

#### Actual behavior
It claims this conflicts with the `unified-signatures` rule because according to tslint, it can be unified to:

```ts
protected oneOf(...accepted: string[] | number[]): (arg1: string, arg2: any) => void;
```

However this is not supported by TypeScript as TypeScript doesn't support this unification.
It gives an error stating that a rest paramter should be of type Array.

#### Expected behavior

Tslint knows this is an impossible unification and so does not mark this in conflict with the `unified-signatures` rule.

Note: TypeScript will most likely never allow this unification as this exact issue (TypeScript part) is marked as `By Design` a.k.a. this is not possible on purpose. (and with a reason)

Typescript issue: https://github.com/Microsoft/TypeScript/issues/5077
